ramips: change sysupgrade method and device name of netis WF-2881

mans0n opened the pull request:
~~- fix LAN/WAN MAC addresses~~
~~- fix LAN port order~~
~~- fix WiFi LEDs~~
~~- fix pinmux groups~~
~~- provide label MAC address~~
~~- switch from gpio-keys-polled to gpio-keys~~
~~- remove nonexistent u-boot-env partition~~

WF-2881 sysupgrade image uses UBI rootfs, but still relies on
default_do_upgrade. Because of this, config backup is not restored after
sysupgrade. It can be fixed by switching to nand_do_upgrade and
sysupgrade-tar image. default_do_upgrade does not handle sysupgrade-tar
properly, so one should use factory image to upgrade from older version.

It means that this commit brings a breaking change. It seems to be a
perfect opportunity to change the device name to the correct one, WF2881
without hyphen.

mans0n committed:
ramips: use nand_do_upgrade for netis WF-2881

WF-2881 sysupgrade image uses UBI rootfs, but still relies on
default_do_upgrade. Because of this, config backup is not restored after
sysupgrade. It can be fixed by switching to nand_do_upgrade and
sysupgrade-tar image. default_do_upgrade does not handle sysupgrade-tar
properly, so one should use factory image to upgrade from older version.

Signed-off-by: Sungbo Eo <mans0n@gorani.run>

mans0n commented:
Removed old boardname from SUPPORTED_DEVICES (inspired by https://github.com/openwrt/openwrt/commit/52f2d7d2a9dc3e142bb957462a30806eb5cebaf4)

adrianschmutzler commented:
> Removed old boardname from SUPPORTED_DEVICES (inspired by 52f2d7d)

But wouldn't you have to rename the board then (as in the reference), because otherwise old devices with netis,wf-2881 compatible could still upgrade, and only those very old ones with wf-2881 couldn't?

mans0n commented:
@adrianschmutzler Yeah that's right, but it's also for stable releases. In 18.06 and 19.07 branch the device uses old wf-2881 boardname, and the boardname change is made recently in snapshot branch. AFAIK breaking changes are acceped in snapshot.

adrianschmutzler commented:
> and the boardname change is made recently in snapshot branch.

Ah, so it was my own board rename patchset from July.

> AFAIK breaking changes are acceped in snapshot.

Well, breaking things is always unpleasant. However, in this case I would prefer it to having to rename this device to something else just because of the change.

mans0n commented:
Hmm..then how about renaming it from wf-2881 to wf2881? Then it would get the correct device name too.

adrianschmutzler commented:
I was arguing in the opposite direction, but if the real name actually is wf2881 then this would be a perfect opportunity to change it.

mans0n committed:
ramips: fix device name of netis WF-2881

The correct model name of WF-2881 is WF2881 without hyphen. The former used
boardnames are not added to SUPPORTED_DEVICES, to make it explicit that the
sysupgrade-tar image, which is newly added in the previous commit, should
not be used to upgrade from older version.

Signed-off-by: Sungbo Eo <mans0n@gorani.run>

mans0n commented:
Added renaming commit. I'm not sure if the commit should come last or first in the series...

adrianschmutzler commented:
I think having it last is okay, after the upgrade change ...

mans0n commented:
Rebased.

mans0n commented:
Rebased due to 19724e2.

adrianschmutzler commented:
@mans0n I plan to merge your first commit here, but beforehand please provide a more extensive discussion of the changes in your commit message. As with the other commits, you should explain what you do and why for every point in your list (obviously you will have more to explain for the partition change than for the label MAC address. Maybe change to a numbered list for that, it might be easier to grasp than the hyphens.

mans0n commented:
@adrianschmutzler Okay, done.

adrianschmutzler commented:
Please also add for 4. and 5. :-)

mans0n commented:
I thought they're quite obvious... Done.

adrianschmutzler commented:
No change without explanation :-)

I've pulled patch 1 into my staging tree:
https://git.openwrt.org/?p=openwrt/staging/adrian.git;a=shortlog;h=refs/heads/staging

adrianschmutzler commented:
To get more attraction for the nand stuff, maybe change PR title and first comment to something concerned with the sysupgrade changes.

mans0n commented:
@adrianschmutzler Thanks for your suggestion. :)

mans0n commented:
Rebased due to 62b1559.

mans0n commented:
Rebased due to e68539a.

mans0n commented:
I wish this is merged before the next 20.xx stable release (and preferably before the branch).
@adrianschmutzler Could you tell me do you like this PR or not? If you think it would be better not to merge this, I'll close it right away.

adrianschmutzler commented:
@mans0n I cannot review the nand_do_upgrade patch well enough to merge it, so you will just have to wait for someone else on that point. The rename patch is fine IMO, but it obviously only makes sense after the other one.

mans0n commented:
@adrianschmutzler I see. I'll try sending it to the mailing list someday. Thanks.

adrianschmutzler commented:
@blocktrron Can you have a look at this?

mans0n commented:
Rebased due to e417ff8.

mans0n commented:
Rebased.

blocktrron reviewed (approved):
LGTM, but needs rebase

If the device cannot keep it's configuration, fixing is not really a breaking change (If you're asking me)

mans0n committed:
ramips: append tail to WF2881 initramfs image

Stock firmware has a vendor-defined tail at the end of uImage for image
validation. This patch enables OpenWrt installation from stock firmware
without having to access the UART console.

Installation via web interface:
1.  Flash **initramfs** image through the stock web interface.
2.  Boot into OpenWrt and perform sysupgrade with sysupgrade image.

Signed-off-by: Sungbo Eo <mans0n@gorani.run>

mans0n commented:
@blocktrron Thanks for the review!

Rebased due to dc862be30192, and also added netis-tail as I did in #2540.

adrianschmutzler commented:
Thanks, I've merged the two patches doing nand and rename. For the netis-tail I think it would be better to open a new PR.

@blocktrron @mans0n Since the nand_do_upgrade looks to me like a fix, would it be reasonable to apply it for 19.07 as well (maybe without the rename, following blocktrron's suggestion)?

mans0n commented:
> For the netis-tail I think it would be better to open a new PR.

Okay, I'll do it.

> would it be reasonable to apply it for 19.07 as well?

IMO this *is* a breaking change, as doing sysupgrade from older version with new sysupgrade-tar image would lead to soft-brick. Introducing such change between point release would bring more confusion to the users, I guess.

adrianschmutzler commented:
> IMO this is a breaking change, as doing sysupgrade from older version with new sysupgrade-tar image would lead to soft-brick. Introducing such change between point release would bring more confusion to the users, I guess.

Ah okay, I wasn't remembering that correctly, so let's just have it in master and we are done. :-)
